docs/source/schemas/dynamic_columns.rst

```diff
@@ -0,0 +1,263 @@
+===============
+Dynamic Columns
+===============
+
+.. note::
+
+    This document discusses AtlasDB's dynamic columns. This is a separate concept from Cassandra's dynamic columns;
+    in fact, AtlasDB's schemas in Cassandra always use two Cassandra dynamic columns, independent of whether
+    we are storing a fixed-columns or dynamic columns table! Throughout this document, where "dynamic columns" is
+    used unqualified, we refer to AtlasDB dynamic columns.
+
+Overview
+--------
+
+AtlasDB offers support for *dynamic columns* - in tables where these are enabled, rows that can have an arbitrary number
+of columns, each with a (typed) column name. A table with dynamic columns may be reasoned about as a
+``Map<RowKey, SortedMap<Tuple<Col1, Col2... ColN>, Value>>``. It may also be thought of as a collection of key-value
+pairs within a row. For example, a simple schema using dynamic columns may be defined as follows:
+
+.. code:: java
+
+    schema.addTableDefinition("todo", new TableDefinition() {{
+                rowName();
+                    rowComponent("person", ValueType.STRING);
+                dynamicColumns();
+                    columnComponent("taskSize", ValueType.VAR_LONG);
+                    columnComponent("monetaryCost", ValueType.VAR_LONG);
+                    value(ValueType.STRING);
+            }});
+
+Note that dynamic column components must be primitive ValueTypes which support partitioning and ordering.
+Also, key-value pairs in an individual row will be returned in *lexicographically* sorted order of the key.
+
+We call the person the *row key*, the (taskSize, monetaryCost) pair the *dynamic column key*, and the string (which is
+intended to be a description of the todo) the *value*.
+
+Query Capabilities
```

Comment:
Yep! Time to dust off my RST skills.

Basically:
- the best cases are those which are contiguous and easily accessible i.e. at the beginning (1, 2 and 8), or in well defined range scans (e.g. 3).
- you pay some penalty for noncontiguous queries e.g. numbers 5 and 6, or if the schema doesn't work because of ordering e.g. 4 and 7.
- row range scans (9) are painful.